base/commands/migration/stages.go

```diff
@@ -92,28 +92,44 @@ func (st *Stages) startStage(ctx context.Context) func(stage.Statuser) error {
 		if err := st.startQueue.Put(ctx, serialization.JSON(b)); err != nil {
 			return err
 		}
+		if err = st.waitStatusResult(ctx, 10, 1*time.Second); err != nil {
+			return err
+		}
 		return nil
 	}
 }
 
 func (st *Stages) migrateStage(ctx context.Context) func(statuser stage.Statuser) error {
 	return func(status stage.Statuser) error {
-		for {
-			s, err := st.readStatus(ctx)
-			if err != nil {
-				return fmt.Errorf("reading status: %w", err)
-			}
-			switch s {
-			case statusComplete:
-				return nil
-			case statusCanceled:
-				return clcerrors.ErrUserCancelled
-			case statusFailed:
-				return errors.New("migration failed")
+		return st.waitStatusResult(ctx, -1, 5*time.Second)
+	}
+}
+
+func (st *Stages) waitStatusResult(ctx context.Context, maxRetries int, period time.Duration) error {
+	tryInfinitely := maxRetries < 0
```

Comment:
I think it would be better to attach a timeout to the context instead of setting max retries.